File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
    
        int[] answer = {0, 0};

        // 올바른 함수 넣기
        for(String key : keyinput){
            switch(key){
                case "up":
                    if(answer[1] + 1 <= board[1] / 2)
                        func1(answer);
                    break;
                case "down":
                    if(answer[1] - 1 >= -(board[1] / 2))
                        func4(answer);
                    break;
                case "right":
                    if(answer[0] + 1 <= board[0] / 2)
                        func3(answer);
                    break;
                default:
                    if(answer[0] - 1 >= -(board[0] / 2))
                        func2(answer);
            }
        }
        
        return answer;
    }
    
    
    private void func1(int[] arr){
        arr[1]++;
    }
    
    private void func2(int[] arr){
        arr[0]--;
    }
    
    private void func3(int[] arr){
        arr[0]++;
    }
    
    private void func4(int[] arr){
        arr[1]--;
    }
    
}

File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    command = {"up": (0, 1), "down": (0, -1), "left": (-1, 0), "right": (1, 0)}
    
    n, m = board
    x, y = 0, 0
    width, height = n >> 1, m >> 1
    
    for input in keyinput:
        dx, dy = command.get(input)
        nx, ny = x + dx, y + dy
        if -(width) <= nx <= width and -(height) <= ny <= height:
            x, y = nx, ny

    return [x, y]